pose_est_nets/losses/losses.py

```diff
@@ -199,6 +200,23 @@ def TemporalLoss(
     loss = loss.masked_fill(mask=loss < epsilon, value=0.0)
     return torch.mean(loss)  # pixels
 
+@typechecked
+def UnimodalLoss(
+    keypoint_preds: TensorType["batch", "two_x_num_keypoints"],
+    heatmap_preds: TensorType["batch", "num_keypoints", "heatmap_height", "heatmap_width"],
+    original_image_height: torch.Tensor,
+    original_image_width: torch.Tensor,
+    output_shape: tuple,
+    **kwargs  # make loss robust to unneeded inputs
+)-> TensorType[(), float]:
+    keypoint_preds = keypoint_preds.reshape(keypoint_preds.shape[0], -1, 2)
+    ideal_heatmaps = generate_heatmaps( #this process doesn't compute gradients
```

Comment:
I just meant that the process of generating the ideal_heatmaps doesn't compute gradients. As you said this isn't necessary because the backpropagation from the loss goes through heatmap_preds to the weights.